core/systems/performance/performance_manager.gd

```diff
@@ -180,88 +185,60 @@ func load_profile(profile_path: String = "") -> void:
 		return
 
 	profile = loaded
-	notify.text = "Loaded performance profile: " + profile.name
+	notify.text = "Loaded performance profile: " + profile_state + " " + profile.name 
 	logger.info(notify.text)
+	logger.debug(profile)
 	_notification_manager.show(notify)
-	await _apply_profile()
+	_shared_thread.exec(_apply_profile)
 
 
-func _get_profile_name() -> String:
-	var prefix: String = _platform.get_product_name().replace(" ", "_") + "_" + profile_state
-	var postfix: String = "_default_profile.tres"
-	if library_item:
-		postfix = "_" + library_item.name.sha256_text() + ".tres"
-		profile.name = library_item.name
-	return prefix+postfix
-
-
-func _update_profile(do_save: bool = true) -> void:
-	if not profile:
-		logger.debug("Profile not set, cannot update.")
-		return
-	if cpu.boost_capable:
-		profile.cpu_boost_enabled = cpu_boost_enabled
-	if cpu.smt_capable:
-		profile.cpu_core_count_current = cpu_core_count_current
-		profile.cpu_smt_enabled = cpu_smt_enabled
-	if gpu.clk_capable:
-		profile.gpu_freq_max_current = gpu_freq_max_current
-		profile.gpu_freq_min_current = gpu_freq_min_current
-		profile.gpu_manual_enabled = gpu_manual_enabled
-	if gpu.power_profile_capable:
-		profile.gpu_power_profile = gpu_power_profile
-	if gpu.tj_temp_capable:
-		profile.gpu_temp_current = gpu_temp_current
-	if gpu.tdp_capable:
-		profile.tdp_boost_current = tdp_boost_current
-		profile.tdp_current = tdp_current
-	if gpu.thermal_profile_capable:
-		profile.thermal_profile = thermal_profile
-	if do_save:
-		save_profile()
+func emit_profile_signals() -> void:
```

Comment:
If we're emitting all this stuff here, why not just have one signal that gets passed the profile?

Reply:
This gets called after all of these signals are linked up by the PowerTools menu. We could do that, but we would need a function in PowerTools that would call all the _update functions and pass in the variables from the profile. This was cleaner.